pkg/controller/sdk/reconciler.go

```diff
@@ -67,24 +68,20 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 
 	// Reconcile this copy of the Source and then write back any status
 	// updates regardless of whether the reconcile error out.
-	obj, err = r.provider.Reconciler.Reconcile(ctx, obj)
-	if err != nil {
-		logger.Warnf("Failed to reconcile %s: %v", r.provider.Parent.GetObjectKind(), err)
+	obj, reconcileErr := r.provider.Reconciler.Reconcile(ctx, obj)
+	if reconcileErr != nil {
+		logger.Warnf("Failed to reconcile %s: %v", r.provider.Parent.GetObjectKind(), reconcileErr)
 	}
 
-	if chg, err := r.statusHasChanged(ctx, original, obj); err != nil || !chg {
-		// If we didn't change anything then don't call updateStatus.
-		// This is important because the copy we loaded from the informer's
-		// cache may be stale and we don't want to overwrite a prior update
-		// to status with this stale state.
-		return reconcile.Result{}, err
-	} else if _, err := r.updateStatus(ctx, request, obj); err != nil {
-		logger.Warnf("Failed to update %s status: %v", r.provider.Parent.GetObjectKind(), err)
-		return reconcile.Result{}, err
+	if needsUpdate := r.needsUpdate(ctx, original, obj); needsUpdate {
+		if _, updateErr := r.update(ctx, request, obj); updateErr != nil {
```

Comment:
Similar reasoning as above. We don't want the error from the reconcile call to be overwritten, so I don't want to use the same variable name. I know that due to being scoped in the if statement, this doesn't actually happen, but if both are called `err` and the ':' is removed from this line, then everything compiles but is logically wrong. I want to prevent that simple change from breaking things.

I suppose I really just want `reconcileErr` and can switch this back to `err` if you strongly prefer it.

Reply:
I prefer, but not strongly... I will let it slide but if I ever pass over this code you know it is changing back :D

Reply:
Note that flattening everything to `err` here will break the code in a subtle way. Do we have a test that verifies the error from `r.provider.Reconciler.Reconcile(...)` is returned so that the tests will break if the error from 73 is dropped?

Reply:
I kept `reconcileErr` and changed this one back to `err` (as it follows the 'normal' Go error semantics of check immediately then throw away).

> Do we have a test that verifies the error from r.provider.Reconciler.Reconcile(...) is returned so that the tests will break if the error from 73 is dropped?

I don't think there are any tests for this SDK yet.